pkgs/development/python-modules/flask-basicauth/default.nix

```
@@ -0,0 +1,35 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, fetchpatch
+, flask
+, lib
+}:
+
+buildPythonPackage rec {
+  pname = "flask-basicauth";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "jpvanhal";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-han0OjMI1XmuWKHGVpk+xZB+/+cpV1I+659zOG3hcPY=";
+  };
+
+  propagatedBuildInputs = [ flask ];
+
+  patches = [
+    (fetchpatch {
+      name = "fix-test-flask-ext-imports.patch";
+      url = "https://github.com/jpvanhal/flask-basicauth/commit/23f57dc1c3d85ea6fc7f468e8d8c6f19348a0a81.patch";
+      sha256 = "sha256-njUYjO0TRe3vr5D0XjIfCNcsFlShbGxtFV/DJerAKDE=";
+    })
+  ];
```

I think it's more logical to add patches right after the source.

```suggestion
    patches = [
    (fetchpatch {
      name = "fix-test-flask-ext-imports.patch";
      url = "https://github.com/jpvanhal/flask-basicauth/commit/23f57dc1c3d85ea6fc7f468e8d8c6f19348a0a81.patch";
      sha256 = "sha256-njUYjO0TRe3vr5D0XjIfCNcsFlShbGxtFV/DJerAKDE=";
    })
  ];
  
  propagatedBuildInputs = [ flask ];
```